### ISSUE TITLE
Set argument for redshift connection/cursor in order to fetch float instances?

### ISSUE THREAD

**Issue by ni-todo-spot** [aws/amazon-redshift-python-driver#95]
## Driver version
2.0.905

## Redshift version
PostgreSQL 8.0.2 on i686-pc-linux-gnu, compiled by GCC gcc (GCC) 3.4.2 20041017 (Red Hat 3.4.2-6.fc3), Redshift 1.0.36224

## Client Operating System
Darwin Nativs-MacBook-Pro.local 21.4.0 Darwin Kernel Version 21.4.0: Mon Feb 21 20:34:37 PST 2022; root:xnu-8020.101.4~2/RELEASE_X86_64 x86_64

## Python version
3.8.7

## Problem description
I was wondering if there is some kind of a flag / setting I can set in order to tell the connector return a numeric result as `float` / `np.float` instead of `decimal.Decimal` instance?

I was using the following raw sql on a cursor instance:
```python
cursor.execute("SELECT 1.0 / 7 AS foo")
df = cursor.fetch_dataframe()
output = df.squeeze()
assert not isinstance(output, decimal.Decimal)
```

Thanks in advance!

**Comment by Brooke-white** [comment]
Hey @ni-todo-spot ,

Thanks for reaching out, this is a good question. At this time, all `NUMERIC` values are mapped to `decimal.Decimal`. `decimal.Decimal` is used for `NUMERIC` values as it offer a higher level of precision than `float`. 

I will chat with the Redshift driver team to see if our Redshift JDBC driver offers such an option, and see if this is something we could support. Thank you for your patience :)

**Comment by Brooke-white** [comment]
Hi @ni-todo-spot ,

Small update -- I've scoped work required for this feature and it looks minimal. I'm working on implementation/testing and anticipate we can include this feature in our next release (scheduled for mid-April). At this time, I think supporting an option for returning a numeric as `float` would be most straight forward.

**Comment by ni-todo-spot** [comment]
> Hi @ni-todo-spot ,
> 
> Small update -- I've scoped work required for this feature and it looks minimal. I'm working on implementation/testing and anticipate we can include this feature in our next release (scheduled for mid-April). At this time, I think supporting an option for returning a numeric as `float` would be most straight forward.

Great news! Thank you very much :)

**Comment by Brooke-white** [comment]
Hi @ni-todo-spot ,

We've just released v2.0.906 of redshift_connector which includes this feature. Please be sure to read the disclaimer about this flag, `numeric_to_float` located in the project [README](https://github.com/aws/amazon-redshift-python-driver#connection-parameters) before using. I've copied it below for your convenience. This disclaimer is just warning of the precision differences between `decimal.Decimal` and `float`, to ensure user is aware of this before use :).

> Specifies if NUMERIC datatype values will be converted from decimal.Decimal to float. By default NUMERIC values are received as decimal.Decimal. Enabling this option is not recommended for use cases which prefer the most precision as results may be rounded. Please reference the Python docs on decimal.Decimal to see the tradeoffs between decimal.Decimal and float before enabling this option.

Please let me know if you have any questions by opening a new issue. Have a nice weekend! :)

**Comment by ni-todo-spot** [comment]
@Brooke-white Wow! 👏 
This is what I've been waiting for :)

Thank you very much 😄
